test/e2e-common.sh

```diff
@@ -262,14 +262,14 @@ function knative_teardown() {
   echo ">> Stopping Knative Eventing"
   for i in "${!UNINSTALL_LIST[@]}"; do
     # We uninstall elements in the reverse of the order they were installed.
-    local YAML="${UNINSTALL_LIST[$(( ${#array[@]} - $i ))]}"
+    local YAML="${UNINSTALL_LIST[$(( ${#UNINSTALL_LIST[@]} - 1 - $i ))]}"
     echo ">> Bringing down YAML: ${YAML}"
     kubectl delete --ignore-not-found=true -f "${YAML}" || return 1
   done
   kubectl delete --ignore-not-found=true namespace "${SYSTEM_NAMESPACE}"
-  wait_until_object_does_not_exist namespaces "${SYSTEM_NAMESPACE}"
+  wait_until_object_does_not_exist namespaces "${SYSTEM_NAMESPACE}" ""
```

Comment:
since we have set -o nounset, it was giving $3: unbound variable error, even though $3 is optional.
https://github.com/knative/eventing/blob/3c2ce5e8e2a92373b5270300a36f80cb0891253d/vendor/knative.dev/hack/library.sh#L216-L224